### PR TITLE
Upgrade to LLVM 11 Stable

### PR DESCRIPTION
**Pull request by Nekotekina**

### CODE REVIEW

File: rpcs3/Emu/CPU/CPUTranslator.cpp [hunk 1]
```diff
@@ -203,9 +203,11 @@ llvm::Constant* cpu_translator::make_const_vector<v128>(v128 v, llvm::Type* t)
 		return llvm::ConstantInt::get(t, llvm::APInt(128, llvm::makeArrayRef(reinterpret_cast<const u64*>(v._bytes), 2)));
 	}
 
-	verify(HERE), t->isVectorTy() && llvm::cast<llvm::VectorType>(t)->getBitWidth() == 128;
+	verify(HERE), t->isVectorTy();
 
 	const auto sct = t->getScalarType();
+	verify(HERE), 128 / t->getScalarSizeInBits() == llvm::cast<llvm::VectorType>(t)->getNumElements();
```
Comment (elad335) [inline rpcs3/Emu/CPU/CPUTranslator.cpp:209]:
```suggestion
	verify(HERE), t->getScalarSizeInBits() * llvm::cast<llvm::VectorType>(t)->getNumElements() == 128;
```

**Comment by Nekotekina** [comment]
It IS compatible with old PPU cache.
And please, while testing, don't delete it, at least backup it.
It may help in cause of promlems ("old cache works, new cache doesn't") plus it may be workaround to said hypothetical problem.

File: rpcs3/Emu/CPU/CPUTranslator.h [hunk 2]
```diff
@@ -2283,9 +2283,11 @@ struct llvm_splat
 
 		if (auto i = llvm::dyn_cast_or_null<llvm::ShuffleVectorInst>(value))
 		{
-			if (llvm::isa<llvm::ConstantAggregateZero>(i->getOperand(2)))
+			if (llvm::isa<llvm::ConstantAggregateZero>(i->getOperand(1)) || llvm::isa<llvm::UndefValue>(i->getOperand(1)))
 			{
-				if (auto j = llvm::dyn_cast<llvm::InsertElementInst>(i->getOperand(0)))
+				static constexpr int zero_array[llvm_value_t<U>::is_vector]{};
```
Comment (elad335) [inline rpcs3/Emu/CPU/CPUTranslator.h:2288]: I think theres a typo here, definitely not a boolean to indicate array size, maybe vector elements count.

Comment (Nekotekina) [reply rpcs3/Emu/CPU/CPUTranslator.h:2288]: is_vector is not boolean, it's a value that has vector number element or zero if it's not a vector. Confusing, but simplifies some things.

**Comment by web1018** [comment]
Should we test this PR for any Possible Performance Improvements?

**Comment by Nekotekina** [comment]
Doubt it, but you can try. We are looking for regressions first.

**Pull request merged**